pylspclient/lsp_structs.py

```diff
@@ -14,50 +15,59 @@ def to_type(o, new_type):
         return new_type(**o)
 
 
-class Position(object):
-    def __init__(self, line, character):
-        """
-        Constructs a new Position instance.
+class Position(BaseModel):
+    """
+    Constructs a new Position instance.
 
-        :param int line: Line position in a document (zero-based).
-        :param int character: Character offset on a line in a document (zero-based).
-        """
-        self.line = line
-        self.character = character
+    :param int line: Line position in a document (zero-based).
+    :param int character: Character offset on a line in a document (zero-based).
+    """
+    line: int
+    character: int
 
 
-class Range(object):
-    def __init__(self, start, end):
-        """
-        Constructs a new Range instance.
+class Range(BaseModel):
+    """
+    Constructs a new Range instance.
 
-        :param Position start: The range's start position.
-        :param Position end: The range's end position.
-        """
-        self.start = to_type(start, Position)
-        self.end = to_type(end, Position)
+    :param Position start: The range's start position.
+    :param Position end: The range's end position.
+    """
+    start: Any
+    end: Any
+    def __init__(self, **data):
+        data["start"] = to_type(data["start"], Range)
+        data["end"] = to_type(data["end"], Range)
+        super().__init__(**data)
 
 
-class Location(object):
+class Location(BaseModel):
     """
     Represents a location inside a resource, such as a line inside a text file.
     """
-    def __init__(self, uri, range):
+    uri: str
+    range: Any
+    def __init__(self, **data):
         """
         Constructs a new Location instance.
 
         :param str uri: Resource file.
         :param Range range: The range inside the file
         """
-        self.uri = uri
-        self.range = to_type(range, Range)
+        data["range"] = to_type(data["range"], Range)
+        super().__init__(**data)
 
 
-class LocationLink(object):
+class LocationLink(BaseModel):
     """
     Represents a link between a source and a target location.
     """
-    def __init__(self, originSelectionRange, targetUri, targetRange, targetSelectionRange):
+    originSelectionRange: Any
+    targetUri: str
+    targetRange: Any
+    targetSelectionRange: Any
+
+    def __init__(self, **data):
```

Comment:
I'm writing it here but it is true for the rest of the classes as well - 
1. What is the usage of `__init__` function in these cases, and not init the members directly?
2. What is the need for the `to_type()` function here? I think it is possible to directly use the Pydantic structs in this case.

Reply:
In the original init definitions, the datatype as being converted. So to preserve that logic, I included the init which would add to the functionality rather than replace. We can of course simply use the following:

class LocationLink(BaseModel):
    originSelectionRange: Range
    targetUri: str
    targetRange: Range
    targetSelectionRange: Range